### DISCUSSION TITLE
async Callback function to make web request from dotnet?

### DISCUSSION THREAD

**Discussion by DanielHabenicht** [elringus/bootsharp#142] [General]
Hey, 

I've only seen examples of non asynchronous function calls from dotnet to javascript. But would love something like this: 
```cs
    [JSFunction] // assigned in JS as `bootsharp.Global.getName = async () => ...`
    public static partial async Task<string> GetFrontendName ();
```

Is it feasible?

**Comment by elringus** [comment]
Here is an example from tests:

https://github.com/elringus/bootsharp/blob/a5cc885f0287a01e91ae38e24437ced53cec22e4/src/js/test/cs/Test/Functions.cs#L21-L22

https://github.com/elringus/bootsharp/blob/a5cc885f0287a01e91ae38e24437ced53cec22e4/src/js/test/spec/interop.spec.ts#L148-L154
